plugins/modules/cloud/misc/terraform.py

```diff
@@ -319,11 +319,25 @@ def remove_workspace(bin_path, project_path, workspace):
     _workspace_cmd(bin_path, project_path, 'delete', workspace)
 
 
-def build_plan(command, project_path, variables_args, state_file, targets, state, plan_path=None):
+def build_plan(command, project_path, variables_args, state_file, targets, state, apply_args, plan_path=None):
     if plan_path is None:
         f, plan_path = tempfile.mkstemp(suffix='.tfplan')
 
-    plan_command = [command[0], 'plan', '-input=false', '-no-color', '-detailed-exitcode', '-out', plan_path]
+    local_command = command.copy()
+
+    plan_command = [command[0], 'plan']
+
+    if state == "planned":
+        for c in local_command[1:]:
+            plan_command.append(c)
+
+    if state == "present":
+        for a in apply_args:
+            local_command.remove(a)
+        for c in local_command[1:]:
+            plan_command.append(c)
+
+    plan_command.extend('-input=false', '-no-color', '-detailed-exitcode', '-out', plan_path)
```

Comment:
```suggestion
    plan_command.extend(['-input=false', '-no-color', '-detailed-exitcode', '-out', plan_path])
```
I think you need a list here, otherwise you end up with `TypeError: extend() takes exactly one argument (5 given)`.

Reply:
Yes you are correct, this would not work.
Thanks for pointing it out!